sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/ResponseValidationPolicyBuilder.java

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.tables;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.function.BiConsumer;
+
+/**
+ * Builder for a policy to do validation of general response behavior.
+ */
+public class ResponseValidationPolicyBuilder {
+    private final Collection<BiConsumer<HttpResponse, ClientLogger>> assertions = new ArrayList<>();
+
+    /**
+     * Builds the {@link HttpPipelinePolicy} described by this builder.
+     *
+     * @return The {@link HttpPipelinePolicy}.
+     */
+    public HttpPipelinePolicy build() {
+        return new ResponseValidationPolicy(assertions);
+    }
+
+    /**
+     * Fluently applies an optional validation to this policy where, if the response contains the given header, asserts
+     * its value is an echo of the value provided in the request.
+     *
+     * @param headerName The header to validate.
+     *
+     * @return The updated {@link HttpPipelinePolicy}.
+     */
+    public ResponseValidationPolicyBuilder addOptionalEcho(String headerName) {
```

Comment:
Just curious if we really need this validation.

Reply:
See my comment above.